urld558a3d988_gao715108023_resourceMonitor_tgz-pJ8-com_sinopec_agent_AgentJ8/src/com/sinopec/agent/infocollect/DynamicNetworkInfo.java

```diff
@@ -0,0 +1,174 @@
+			SharedArea.dynamicInfoSerializable.setOutDatagramsPs(outDatagramsPs);
+		}
+	}
+
+	/**
+	 * 从/proc/net/snmp中获取TCP/UDP/IP的信息
+	 * 
+	 * @return mind
+	 * @throws IOException
+	 */
+	private double[] getCommondTCPUDPIP() throws IOException {
+		double mind[] = new double[8];
+		BufferedReader buffer = null;
+		try {
+			buffer = new BufferedReader(new FileReader("/proc/net/snmp"));
+			String line = null;
+			int ipnum = 0, ipn = 0;
+			int tcpnum = 0, tcpn = 0;
+			int udpnum = 0, udpn = 0;
+			while ((line = buffer.readLine()) != null) {
+				if (line.contains("Ip:") && ipnum == 0) {
+					ipnum++;
+				} else if (line.contains("Ip:") && ipnum == 1) {
+					ipnum++;
+					StringTokenizer toker = new StringTokenizer(line);
+					while (toker.hasMoreElements()) {
+						toker.nextToken();
+						ipn++;
+						if (ipn == 3) {
+							mind[0] = Double.parseDouble(toker.nextToken().trim());
+							ipn++;
+						} else if (ipn == 9) {
+							mind[1] = Double.parseDouble(toker.nextToken().trim());
+							ipn++;
+							mind[2] = Double.parseDouble(toker.nextToken().trim());
+							ipn++;
+							break;
+						}
+					}
+				}
+
+				if (line.contains("Tcp:") && tcpnum == 0) {
+					tcpnum++;
+				} else if (line.contains("Tcp:") && tcpnum == 1) {
+					tcpnum++;
+					StringTokenizer toker = new StringTokenizer(line);
+					while (toker.hasMoreElements()) {
+						toker.nextToken();
+						tcpn++;
+						if (tcpn == 10) {
+							mind[3] = Double.parseDouble(toker.nextToken().trim());
+							mind[4] = Double.parseDouble(toker.nextToken().trim());
+							mind[5] = Double.parseDouble(toker.nextToken().trim());
+							break;
+						}
+					}
+				}
+				if (line.contains("Udp:") && udpnum == 0) {
+					udpnum++;
+				} else if (line.contains("Udp:") && udpnum == 1) {
+					udpnum++;
+					StringTokenizer toker = new StringTokenizer(line);
+					while (toker.hasMoreElements()) {
+						toker.nextToken();
+						udpn++;
+						if (udpn == 1) {
+							mind[6] = Double.parseDouble(toker.nextToken().trim());
+							udpn++;
+						} else if (udpn == 4) {
+							mind[7] = Double.parseDouble(toker.nextToken().trim());
+							udpn++;
+							break;
+						}
+					}
+				}
+			}
+		} catch (FileNotFoundException e) {
+			// TODO Auto-generated catch block
+			LOG.error(LogUtils.getTrace(e));
+		} catch (IOException e) {
+			// TODO Auto-generated catch block
+			LOG.error(LogUtils.getTrace(e));
+		} finally {
+			if (buffer != null)
+				buffer.close();
+		}
+		return mind;
+	}
+
+	@Override
+	public void run() {
+		// To change body of implemented methods use File | Settings | File
+		// Templates.
+		while (running) {
+			try {
+				getNetworkInfoByProc();
+			} catch (IOException e) {
+				e.printStackTrace(); // To change body of catch statement use
+										// File | Settings | File Templates.
+			}
+			try {
+				Thread.sleep(samplingInterval);
+			} catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)